pallets/circuit/src/lib.rs

```diff
@@ -1430,51 +1103,30 @@ impl<T: Config> Pallet<T> {
                     Self::deposit_event(Event::XTransactionStepFinishedExec(xtx_id)),
                 CircuitStatus::FinishedAllSteps =>
                     Self::deposit_event(Event::XTransactionXtxFinishedExecAllSteps(xtx_id)),
-                CircuitStatus::RevertTimedOut =>
+                CircuitStatus::Reverted(ref _cause) =>
                     Self::deposit_event(Event::XTransactionXtxRevertedAfterTimeOut(xtx_id)),
-                CircuitStatus::DroppedAtBidding =>
+                CircuitStatus::Killed(ref _cause) =>
                     Self::deposit_event(Event::XTransactionXtxDroppedAtBidding(xtx_id)),
                 _ => {},
             }
-            if xtx.status >= CircuitStatus::PendingExecution {
+            if xtx.status.clone() >= CircuitStatus::PendingExecution {
                 if let Some(full_side_effects) = maybe_full_side_effects {
                     Self::deposit_event(Event::SideEffectsConfirmed(xtx_id, full_side_effects));
                 }
             }
         }
     }
 
-    fn kill(local_ctx: &mut LocalXtxCtx<T>, cause: CircuitStatus) {
-        local_ctx.xtx.status = cause.clone();
-
-        match cause {
-            CircuitStatus::RevertTimedOut => {
-                if let Err(err) = Optimistic::<T>::try_slash(local_ctx) {
-                    log::error!(target: "circuit", "Failed to slash {:?} for xtx {:?}", err, local_ctx.xtx_id);
-                }
-            },
-            CircuitStatus::DroppedAtBidding => {
-                Optimistic::<T>::try_dropped_at_bidding_refund(local_ctx);
-            },
-            _ => {},
-        }
-
-        Self::square_up(local_ctx, None)
-            .expect("Expect Revert and RevertKill options to square up to be infallible");
-
-        Self::apply(local_ctx, (cause.clone(), cause));
-    }
-
     fn square_up(
```

Comment:
This function im a bit dubious of. Wouldn't it make sense to tie the different operations/updates directly into the transitions of the state machine? The we would only keep the transitions there in mind when updating and we can modularize the logic here.